crates/bevy_pbr/src/meshlet/material_draw_prepare.rs

```diff
@@ -139,8 +139,8 @@ pub fn prepare_material_meshlet_meshes_main_opaque_pass<M: Material>(
 
         view_key |= MeshPipelineKey::from_primitive_topology(PrimitiveTopology::TriangleList);
 
-        for material_id in render_material_instances.values() {
-            let Some(material) = render_materials.get(*material_id) else {
+        for &material_key in render_material_instances.values() {
```

Comment:
I have this fixed in a future PR by changing it to `.values().collect::<HashSet<_>>()`, but this code is currently bugged and adds too many materials, as it does not deduplicate between multiple entities with the same material (AssetId).

What this code really wants to do is iterate over every instance of material M that exists. I don't think we have a good way of doing this, even with these changes, right?